src/lib/hook-mutation.ts

```diff
@@ -0,0 +1,33 @@
+// @ts-ignore
+import { useState, useEffect, useContext } from 'react';
+import { context } from '../components/context';
+import { Client, ExchangeResult, MutationHook } from '../types';
+
+export function useMutation<Args>(
+  mutationQuery: string,
+  passedVars: Object = {}
+): MutationHook<Args> {
+  const client = (useContext(context) as any) as Client;
+
+  return variables =>
+    new Promise((resolve, reject) => {
+      const inst = client.createInstance({
+        onChange: result => {
+          if (result.error) {
+            reject(result.error);
+            return;
+          }
+
+          resolve(result);
+        },
+      });
+
+      inst.executeMutation({
+        query: mutationQuery,
+        variables: {
+          ...passedVars,
+          ...(variables as Object),
```

Comment:
Are you sure we want to use capital O "Object" here? I think you probably want small o `object`: https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html